src/node.cpp

```diff
@@ -1364,6 +1364,11 @@ void LocalNode::init(Sync* csync, nodetype_t ctype, LocalNode* cparent, string*
     if (cparent)
     {
         setnameparent(cparent, cfullpath);
+
+        if (parent->node)
+        {
+            h = parent->node->nodehandle;
```

Comment:
I think we should do this assignment at megaclient.cpp:13151  instead, at the moment the PUT transfer is started, to ensure we pick the up to date handle (as actionpackets may later rearrange the parent pointers etc). thanks!

Reply:
I agree. Thanks, Matt ;)